examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py

```diff
@@ -1778,7 +1848,8 @@ def compute_text_embeddings(prompt, text_encoders, tokenizers):
     if freeze_text_encoder and not train_dataset.custom_instance_prompts:
         del tokenizers, text_encoders
         gc.collect()
-        torch.cuda.empty_cache()
+        if torch.cuda.is_available():
+            torch.cuda.empty_cache()
```

Comment:
torch.mps has an equivalent to this, if you're interested. though it seems out of scope for these adjustments